### PR TITLE
Added External DNS Feature

### PR DESCRIPTION
**Pull request by mohith10**
## Description

This pull request integrates the External DNS feature into the product. The following things will be introduced in the PR:

1. External DNS data source.
2. Inventory DNS asset data source.
3. Inventory SRV Protocol data source.
4. External DNS resource.
5. Appropriate additions to the docs and examples directory.

**Note**: The _getExternalDNSRecord_ call doesn't contain 'dns asset' information in the response, contrary to what is expected. Resulting in a 400 Bad Request when the 'dns asset id' is sent in the POST request. Eduardo Lemes is working this issue. So this is being sidelined for this PR and will be fixed in the future.

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

File: docs/data-sources/inventory_srv_protocols.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+- `id` (String) The ID of this resource.
+- `inventory_srv_protocols` (List of Object) (see [below for nested schema](#nestedatt--inventory_srv_protocols))
+
+<a id="nestedatt--inventory_srv_protocols"></a>
+### Nested Schema for `inventory_srv_protocols`
+
+Read-Only:
+
+- `id` (String)
+- `protocols` (String)
```
Comment (hughwphamill) [inline docs/data-sources/inventory_srv_protocols.md:30]: I wonder why this is being pluralized?

**Comment by hughwphamill** [comment]
> Note: The getExternalDNSRecord call doesn't contain 'dns asset' information in the response, contrary to what is expected. Resulting in a 400 Bad Request when the 'dns asset id' is sent in the POST request. Eduardo Lemes is working this issue. So this is being sidelined for this PR and will be fixed in the future.

Can you explain this a little more, is the feature broken until the future fix?

File: docs/data-sources/inventory_srv_protocols.md (hunk 1 above)
Comment (mohith10) [reply docs/data-sources/inventory_srv_protocols.md]: Oops. Fixed this typo.

**Comment by mohith10** [comment]
> > Note: The getExternalDNSRecord call doesn't contain 'dns asset' information in the response, contrary to what is expected. Resulting in a 400 Bad Request when the 'dns asset id' is sent in the POST request. Eduardo Lemes is working this issue. So this is being sidelined for this PR and will be fixed in the future.
> 
> Can you explain this a little more, is the feature broken until the future fix?

I suspect it as an issue on the API server side.

So when I am adding the **_dnsAssets_** info in my request, it is throwing "Forbidden" error.
image

Also if i make GET request to fetch the external DNS Assets, the dnsAsset field isn't there.
image

**Review by hughwphamill — APPROVED** [review]

**Pull request merged**